### PR TITLE
build.cabal.sh: Use cabal new-run

### PR DESCRIPTION
**Pull request by bgamari**
The previous approach was terribly unreliable, leading me to waste an hour
debugging #425.

### CODE REVIEW

File: build.cabal.sh [hunk 1]
```diff
@@ -48,7 +48,7 @@ if [ "${CABVER[0]}" -eq 2 -o "${CABVER[0]}" -eq 1 -a "${CABVER[1]}" -ge 24 ]; th
 
     # there's no 'cabal new-run' yet, but it's easy to emulate
```
Comment (snowleopard) [inline build.cabal.sh]: Shall this comment be dropped then?

**Comment by snowleopard** [comment]
@bgamari Thanks! Travis failed though: https://travis-ci.org/snowleopard/hadrian/jobs/286173747

Travis uses `build.cabal.sh`, so it looks like this change broke it?

**Comment by bgamari** [comment]
Fixed.

Also note that the change in #385 wasn't quite right, which I fixed in the second commit. However, I would be happy to move this to a new PR if you would prefer.

**Comment by snowleopard** [comment]
@bgamari I'm fine to keep the `rtsopts` bit in this PR.

However, I don't see the commit fixing the `new-run` problem with the CI -- it still fails!

**Comment by bgamari** [comment]
Huh, odd; my impression that the reason for the previous failure was that I was missing a `--` in the command line. Apparently not.

**Comment by bgamari** [comment]
Hmm, I wonder if this is due to an outdated `cabal-install`.

**Comment by bgamari** [comment]
Alright, well until I get a chance to sort this out I've split out the rts options fix into #429.

**Comment by snowleopard** [comment]
@bgamari Apologies for messing up the PR with a conflicting commit (#430).

**Comment by bgamari** [comment]
Continuing in #435.

**Pull request closed without merging**